implementation/src/main/java/io/smallrye/faulttolerance/HystrixCommandInterceptor.java

```diff
@@ -190,18 +212,107 @@ private Object executeCommand(Function<Supplier<Object>, SimpleCommand> commandF
                         syncCircuitBreaker.executionSucceeded();
                     }
                 }
+                //Metrics integration
+                if (retryContext != null) {
+                    if (retryContext.hasBeenRetried()) {
+                        counterInc(prefix + ".retry.callsSucceededRetried.total");
+                    } else {
+                        counterInc(prefix + ".retry.callsSucceededNotRetried.total");
+                    }
+                }
+                if (metadata.operation.hasTimeout()) {
+                    counterInc(prefix + ".timeout.callsNotTimedOut.total");
+                }
+                if (metadata.operation.hasCircuitBreaker()) {
+                    counterInc(prefix + ".circuitbreaker.callsSucceeded.total");
+
+                }
+
+
                 return res;
             } catch (HystrixRuntimeException e) {
-                Exception res = processHystrixRuntimeException(e, retryContext, metadata.operation.getMethod(), syncCircuitBreaker);
-                if (res != null) {
-                    throw res;
+
+                if (metadata.operation.hasCircuitBreaker()) {
+                    counterInc(prefix + ".circuitbreaker.callsFailed.total");
+                    if(command.isCircuitBreakerOpen()) {
+                        counterInc(prefix + ".circuitbreaker.callsPrevented.total");
+                    }
+                }
+
+                //Metrics integration
+                if(e.getFallbackException() != null &&(retryContext == null || retryContext.isLastAttempt())) {
+                    counterInc(prefix + ".fallback.calls.total");
                 }
+                //Metrics integration
+                if (retryContext != null ) {
+                    if (retryContext.isLastAttempt()) {
+                        counterInc(prefix + ".retry.callsFailed.total");
+                        counterInc(prefix + ".invocations.failed.total");
+                    }
+                } else {
+                    counterInc(prefix + ".invocations.failed.total");
+                }
+
+
+                exception = processHystrixRuntimeException(e, retryContext, metadata.operation.getMethod(), syncCircuitBreaker);
+                if (exception != null) {
+                    if(exception.getClass() == TimeoutException.class) {
+                        counterInc(prefix + ".timeout.callsTimedOut.total");
+                    }
+                    throw exception;
+                }
+            } finally {
+                if(start != 0) {
+                    histogramUpdate(prefix + ".timeout.executionDuration", System.nanoTime() - start);
+                }
+                if(command.isResponseFromFallback()) {
+                    counterInc(prefix + ".fallback.calls.total");
+                }
+
             }
         }
     }
 
+    private void counterInc(String name) {
+        //TODO:disable metrics by config
```

Comment:
I think this is essential... not all users are interested in metrics.

Reply:
It is now integrated